docs/platform_ccl.rst

```diff
@@ -0,0 +1,297 @@
+
+Qubits available in the pltform are specified in ``qubits`` section, as shown below. For CC-Light only ``count`` needs to be specified which indiates the number of available qubits.
+
+.. code-block::
+   :linenos:
+
+	"qubits":
+	{
+	    "count": 7
+	},
+
+An instruction may require a wave form generator resource. Waveform generators and their constraints are specified in ``qwgs`` section, as shown below.
+
+.. code-block::
+   :linenos:
+
+	"qwgs" :
+	{
+	  "count": 3,
+	  "connection_map":
+	  {
+	    "0" : [0, 1],
+	    "1" : [2, 3, 4],
+	    "2" : [5, 6]
+	  }
+	},
+
+The number of these waveform generators is specified by the ``count`` field. In the ``connection_map`` it is specified which wafeform generator is connected to which qubits. For instance, Line 6 specifies that ``qwg 0`` is connected to qubits 0 and 1. This is based on ``AWG-8 1, channel 0`` entry in 4th row in Table :numref:`table_ccl_connections` This information is utilized by the scheduler to perform resource-constraint aware scheduling of instructions.
+
+.. note::
+	By providing an empty list for a qwg will result in not applying any qwg constraint during scheduling.
+
+Available measurement/readout units are specified in ``meas_units`` section, as shown below.
+
+.. code-block::
+   :linenos:
+
+	"meas_units" :
+	{
+	  "count": 2,
+	  "connection_map":
+	  {
+	    "0" : [0, 2, 3, 5, 6],
+	    "1" : [1, 4]
+	  }
+	},
+
+
+.. _fig_qubit_numbering_ccl:
+
+.. figure:: ./qubit_number.png
+    :width: 800px
+    :align: center
+    :alt: Qubit numbering and connectivity of qubits in CC-Light Platform
+    :figclass: align-center
+
+    Qubit numbering and connectivity in CC-Light Platform.
+
+
+For the details below, it will be convinient to consider the 
+Figure :numref:`fig_qubit_numbering_ccl`, which shows qubit and edge numbering in CC-Light platform.
+
+
+Edges and the constraints imposed by these edges are specified in ``edges`` section.
+``count`` at Line 3 specifies the number of edges between qubits in the platform.
+``connection_map`` specifies connections. For example, on Line 6, Edge 0 implies a
+constraint on Edge 2 and Edge 10. This means, if Edge 0 is reserved for an operation,
+an operation on Edge 2 and Edge 10 will not be scheduled, until operation on Edge 0
+is complete.
+
+.. code-block::
+   :linenos:
+
+	"edges":
+	{
+	  "count": 16,
+	  "connection_map":
+	  {
+	    "0": [2, 10], 
+	    "1": [3, 11],
+	    "2": [0, 8],
+	    "3": [1, 9],
+	    "4": [6, 14],
+	    "5": [7, 15],
+	    "6": [4, 12],
+	    "7": [5, 13],
+	    "8": [2, 10],
+	    "9": [3, 11],
+	    "10": [0, 8],
+	    "11": [1, 9],
+	    "12": [6, 14],
+	    "13": [7, 15],
+	    "14": [4, 12],
+	    "15": [5, 13]
+	  }
+	},
+
+
+Detunning constraints are specified ``detuned_qubits``. A two-qubit flux gate
+lowers the frequency of its source qubit to get near the frequency of its target
+qubit.  Any two qubits which have near frequencies execute a two-qubit flux gate.
```

Comment:
`Any two qubits which have near frequencies execute a two-qubit flux gate.` is a bit confusing...
Do you mean any two qubits who share an edge can be part of a two-qubit gate? If this is what you mean, the english of the sentence should be changed, to convey that...